backend/tests/Feature/GetAvgSessionTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Tests\Feature;
+
+use Tests\TestCase;
+use Illuminate\Foundation\Testing\WithFaker;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use App\Entities\User;
+use App\Entities\Website;
+use App\Entities\Visitor;
+use App\Entities\Session;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Carbon;
+use App\Repositories\EloquentSessionRepository;
+use Tymon\JWTAuth\Facades\JWTAuth;
+
+class GetAvgSessionTest extends TestCase
+{
+    use RefreshDatabase;
+
+    public function test_average_session_duration()
+    {
+        $user = factory(User::class)->create();
+        $website = factory(Website::class)->create();
+        $user->website()->save($website);
+        $token = JWTAuth::fromUser($user);
+        $websiteVisitors = factory(Visitor::class, 10)->create(['website_id' => $website->id]);
+        $start_session = Carbon::now();
+
+
+        $geoPositionId = factory(\App\Entities\GeoPosition::class)->create()->id;
+        $oSId = factory(\App\Entities\Os::class)->create()->id;
+        $browserId = factory(\App\Entities\Browser::class)->create()->id;
+        $pageId = factory(\App\Entities\Page::class)->create(['website_id' => $website->id])->id;
+        $demographicId = factory(\App\Entities\Demographic::class)->create(['geo_position_id' => $geoPositionId])->id;
+        $deviceId = factory(\App\Entities\Device::class)->create()->id;
+        $systemId = factory(\App\Entities\System::class)->create([
+            'browser_id' => $browserId,
+            'os_id' => $oSId
+        ])->id;
+
+        factory(Session::class, 30)->create([
+            'visitor_id' => $websiteVisitors->pluck('id')->random(),
+            'start_session' => $start_session->addDays(rand(0, 4))->timestamp,
```

Comment:
oh no, never use random in the test. The test must be repeatable, if the test will fail in some cases you will spend a lot of time to find why the test failing. The solution is don't use so many sessions. Create 4 or 5 sessions, some should be out of date range, some in range. And then compare with constant value result

Reply:
And pay attention, that some of the sessions shouldn't belong to website of user